source/shadow.c

```diff
@@ -466,37 +463,34 @@ ShadowStatus_t Shadow_MatchTopic( const char * pTopic,
                                           topicLength - consumedTopicLength,
                                           SHADOW_PREFIX,
                                           SHADOW_PREFIX_LENGTH );
+    }
 
-        if( shadowStatus == SHADOW_SUCCESS )
-        {
-            consumedTopicLength += SHADOW_PREFIX_LENGTH;
+    if( shadowStatus == SHADOW_SUCCESS )
+    {
+        consumedTopicLength += SHADOW_PREFIX_LENGTH;
 
-            /* If no more topic string is left to parse, fail. */
-            if( consumedTopicLength >= topicLength )
-            {
-                shadowStatus = SHADOW_THINGNAME_PARSE_FAILED;
-                LogDebug( ( "Not related to Shadow, thing name is not in pTopic %s, failed to parse thing name", pTopic ) );
-            }
-        }
-        else
+        /* If no more topic string is left to parse, fail. */
+        if( consumedTopicLength >= topicLength )
         {
-            LogDebug( ( "Not related to Shadow, failed to parse shadow topic prefix in pTopic %s", pTopic ) );
+            shadowStatus = SHADOW_THINGNAME_PARSE_FAILED;
+            LogDebug( ( "Not related to Shadow, thing name is not in pTopic %s, failed to parse thing name", pTopic ) );
```

Comment:
These should be LogError()?

Reply:
per offline conversation, it is not an error to fail to parse.

Reply:
We have the status code to notify users that this is not a shadow topic.
PARSE fail means this is not a shadow topic, it doesn't means there is a error that needs to be handle